caas/kubernetes/provider/statefulsets.go

```diff
@@ -109,8 +104,9 @@ func (k *kubernetesClient) ensureStatefulSet(spec *apps.StatefulSet, existingPod
 	if err != nil {
 		return errors.Trace(err)
 	}
-	// TODO(caas) - allow extra storage to be added
 	existing.Spec.Replicas = spec.Spec.Replicas
+	// TODO(caas) - allow storage `request` configurable - currently we only allow `limit`.
+	existing.Spec.VolumeClaimTemplates = spec.Spec.VolumeClaimTemplates
```

Comment:
make sure to test that upgrade charm works as a mistake here can break upgrades

Reply:
Just tested ok.
But I am removing `VolumeClaimTemplates` here because we don't allow change storage, so there will always no change.